docs/proposals/multi-arch-image-mgmt.md

```diff
@@ -0,0 +1,305 @@
+
+```console
+$ oras manifest index create layout-dir:v1 v1-linux-amd64 v1-linux-arm64 --oci-layout --annotation "com.example.key=value" 
+
+Fetching  v1-linux-amd64 
+Fetched   sha256:42c524c48e0672568dbd2842d3a0cb34a415347145ee9fe1c8abaf65e7455b46 v1-linux-amd64 
+Fetching  v1-linux-arm64
+Fetched   sha256:965945e1a08031a63d5970c1da7c39af231c36e4c0a5a3cc276d02a3e06513ee v1-linux-arm64 
+Packed    edb5bc1f0b5c application/vnd.oci.image.index.v1+json
+Pushed    [oci-layout] layout-dir:v1
+Digest: sha256:edb5bc1f0b5c21e9321b34e50c92beae739250fb88409056e8719d9759f6b5b4
+```
+
+3. View the image index in the OCI image layout in a pretty JSON output: 
+
+```bash
+$ oras manifest fetch layout-dir:v1 --oci-layout --pretty  
+```
+
+```json
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.index.v1+json",
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:42c524c48e0672568dbd2842d3a0cb34a415347145ee9fe1c8abaf65e7455b46",
+      "size": 1239,
+      "platform": {
+        "architecture": "amd64",
+        "os": "linux"
+      }
+    },
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:965945e1a08031a63d5970c1da7c39af231c36e4c0a5a3cc276d02a3e06513ee",
+      "size": 1239,
+      "platform": {
+        "architecture": "arm64",
+        "os": "linux"
+      }
+    }
+  ],
+  "annotations": {
+    "com.example.key": "value"
+  }
+}
+```
+
+#### Create a multi-arch image from a registry only
+
+Create a multi-arch image using two arch-specific images stored in a registry only, tag it with `v1` and push the tagged image index to a registry automatically.
+
+```console
+$ oras manifest index create localhost:5000/example:v1 v1-linux-arm64 v1-linux-arm64
+
+Fetching  v1-linux-amd64 
+Fetched   sha256:42c524c48e0672568dbd2842d3a0cb34a415347145ee9fe1c8abaf65e7455b46 v1-linux-amd64 
+Fetching  v1-linux-arm64
+Fetched   sha256:965945e1a08031a63d5970c1da7c39af231c36e4c0a5a3cc276d02a3e06513ee v1-linux-arm64 
+Packed    edb5bc1f0b5c application/vnd.oci.image.index.v1+json
+Pushed    localhost:5000/example:v1
+Digest: sha256:edb5bc1f0b5c21e9321b34e50c92beae739250fb88409056e8719d9759f6b5b4
+```
+
+#### Create a multi-arch image from local and registry
+
+Users need to pull the target arch-specific image to the local repository or OCI image layout, then use `oras manifest index create` to create a multi-arch image.  
+
+Each arch-specific image is supposed to be stored in the same repository to assemble a multi-arch image. The image index acts as a pointer to different arch-specific images, and most OCI registries do not support cross-repository image references in an image index. Keeping all variants in the same repository ensures the registry can resolve and serve the correct architecture. 
+
+```console
+$ oras copy --to-oci-layout ghcr.io/example:v1-linux-amd64 example:v1-linux-amd64
```

Comment:
This is inefficient and we can improve it in the next iteration or milestone in a roadmap.

Reply:
Creating indexes can be viewed as creating new metadata from existing metadata. In other words, we don't need to download gigabytes of blobs from the registry to the local file system. We only need the metadata (i.e. the manifest) of those existing remote images.

Therefore, we should consider a new copy functionality, `oras copy --shallow`. /cc @Wwwsylvia

Reply:
`oras copy --shollow` sounds a bit strange to me. If we don't copy the whole artifact, there would be issue when we copy the created index to other places.

Reply:
@shizhMSFT Do we want to create an issue to track this improvement idea of `oras copy --shallow`? We can discuss whether it worth introducing a new flag in future iteration.

Reply:
@FeynmanZhou Let's discuss offline first for some basic knowledge

Reply:
@shizhMSFT I created a new issue to track this idea: https://github.com/oras-project/oras/issues/1654. I will mark this comment as resolved since no changes required here.